packages/world/src/modules/core/implementations/CallBatchSystem.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import { System } from "../../../System.sol";
+import { IBaseWorld } from "../../../interfaces/IBaseWorld.sol";
+import { revertWithBytes } from "../../../revertWithBytes.sol";
+
+contract CallBatchSystem is System {
+  /**
+   * Call multiple systems at the given resourceSelectors, on behalf of _msgSender().
+   */
+  function callBatch(
+    bytes32[] memory resourceSelectors,
+    bytes[] memory callDatas
```

Comment:
same question as before: would it be better or worse gas-wise to have an array of `SystemCall` structs instead of two arrays? I assume a little worse on calldata but we wouldn't have to do a length check on the arrays to make sure they match (though I guess the whole thing would revert with an out of bounds error in that case)

Reply:
my feeling is it would be a little _better_ calldata wise because the structs are encoded like tuples, so there shouldn't be overhead on the encoding of the individual elements, but we might save one word because we only have one array (and therefore only one length encoding)

Reply:
Going to implement the struct for the calldata improvements and to improve usability (only dealing with a single array)